src/main/java/sirius/biz/jupiter/IDBTable.java

```diff
@@ -0,0 +1,563 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jupiter;
+
+import redis.clients.jedis.Client;
+import sirius.kernel.commons.Limit;
+import sirius.kernel.commons.PullBasedSpliterator;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Values;
+import sirius.kernel.nls.NLS;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Wraps an InfoGraphDB table of an attached Jupiter instance.
+ */
+public class IDBTable {
+
+    private static final JupiterCommand CMD_LOOKUP = new JupiterCommand("IDB.LOOKUP");
+    private static final JupiterCommand CMD_ILOOKUP = new JupiterCommand("IDB.ILOOKUP");
+    private static final JupiterCommand CMD_QUERY = new JupiterCommand("IDB.QUERY");
+    private static final JupiterCommand CMD_IQUERY = new JupiterCommand("IDB.IQUERY");
+    private static final JupiterCommand CMD_SEARCH = new JupiterCommand("IDB.SEARCH");
+    private static final JupiterCommand CMD_ISEARCH = new JupiterCommand("IDB.ISEARCH");
+    private static final JupiterCommand CMD_SCAN = new JupiterCommand("IDB.SCAN");
+    private static final JupiterCommand CMD_ISCAN = new JupiterCommand("IDB.ISCAN");
+
+    private static final int FETCH_PAGE_SIZE = 250;
+
+    private final JupiterConnector jupiter;
+    private final String name;
+
+    /**
+     * Creates a new wrapper for a given Jupiter instance and table name.
+     *
+     * @param jupiter the jupiter connection to wrap
+     * @param name    the name of the table to wrap
+     */
+    public IDBTable(JupiterConnector jupiter, String name) {
+        this.jupiter = jupiter;
+        this.name = name;
+    }
+
+    /**
+     * Used as fluent tool to build queries.
+     */
+    public class QueryBuilder {
+
+        private boolean exact;
+        private String searchPaths;
+        private String searchValue;
+        private String primaryLang;
+        private String fallbackLang;
+
+        /**
+         * Specifies the paths to look for a given value.
+         * <p>
+         * Note that this will perform an exact search (the search value has to match the value in the field).
+         *
+         * @param paths one or more paths to search in
+         * @return the builder itself for fluent method calls
+         */
+        public QueryBuilder lookupPaths(String... paths) {
```

Comment:
To ensure exactly one of the three modes is used at the same time these could be declared as static functions that are the only ways to get an instance of the builder.

Reply:
no you can skip searching entirely and retrieve all rows...

Reply:
okay +1